modules/carousel/jetpack-carousel.php

```diff
@@ -115,7 +134,7 @@ function display_bail_message( $output= '' ) {
 		return $output;
 	}
 
-	function enqueue_assets( $output ) {
+	function check_and_enqueue_assets( $output ) {
```

Comment:
These checks test if other plugin is not processing `[gallery]` shortcode already. If so carousel is not added. However at this moment it is not properly handled for pages that combine both gallery handled by other plugin and single images linking to lightbox. This was mentioned in 2 of my comments above and will be fixed.